plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/libraryElement/impl/ConnectionAnnotations.java

```diff
@@ -352,6 +353,35 @@ private ConnectionAnnotations() {
 		throw new UnsupportedOperationException("Helper class must not be instantiated"); //$NON-NLS-1$
 	}
 
+	public static boolean validateNegatedConnection(final Connection connection, final DiagnosticChain diagnostics,
+			final Map<Object, Object> context) {
+		if (connection == null || connection.getSource() == null || connection.getDestination() == null) {
+			return false;
+		}
+
+		if (connection.isNegated()
+				&& (!connection.getSource().getType().equals(IecTypes.ElementaryTypes.BOOL)
+						|| !connection.getDestination().getType().equals(IecTypes.ElementaryTypes.BOOL))
+				&& !hasInOutVar(connection)) {
```

Comment:
I believe this should be:
```suggestion
				&& (!connection.getSource().getType().equals(IecTypes.ElementaryTypes.BOOL)
						|| !connection.getDestination().getType().equals(IecTypes.ElementaryTypes.BOOL)
						|| hasInOutVar(connection))) {
```

Reply:
Shouldn't all three conditions be met for the error marker to appear? And there **shouldn't** be any inout vars since you removed the '!', or have I misunderstood that?

Reply:
My original comment may have been misleading. Negated connections are *not supported* for inout vars, so there *should* be an error marker if either source or destination is an inout var.